### PR TITLE
#d2l  add  11_10 11_11 13_1 13_2 14_7

### PR DESCRIPTION
**Pull request by Junxiao-Ma**
[11_10 adam.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8016633/11_10.adam.md)
[11_11 lr-scheduler.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8016634/11_11.lr-scheduler.md)

### CODE REVIEW

File: Dive-into-DL-paddlepaddle/docs/11_optimization_algorithm/11_11 lr-scheduler.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        return self.lr * pow(num_update + 1.0, -0.5)
+```
+
+让我们在一系列值上绘制它的行为。
+
+
+
+```python
+scheduler = SquareRootScheduler(lr=0.1)
+d2l.plot(paddle.arange(num_epochs), [scheduler(t) for t in range(num_epochs)])
```
Comment (an1018) [inline Dive-into-DL-paddlepaddle/docs/11_optimization_algorithm/11_11 lr-scheduler.md:142]: 参考最新paddle.py修改paddle.*为d2l.*，麻烦全文自检+修改。同理修改Adam小节，腾讯文档中Adam aistudio链接不对，修改之后生成新版本，更新链接即可

**Comment by Junxiao-Ma** [comment]
[11_11 lr-scheduler.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8029471/11_11.lr-scheduler.md)
[11_10 adam.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8029473/11_10.adam.md)

**Comment by Junxiao-Ma** [comment]
[11_11 lr-scheduler.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8029471/11_11.lr-scheduler.md)
[11_10 adam.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8029473/11_10.adam.md)

**Comment by Junxiao-Ma** [comment]
[11_10 adam.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8049368/11_10.adam.md)
[11_11 lr-scheduler.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8049370/11_11.lr-scheduler.md)

**Comment by Junxiao-Ma** [comment]
[11_11 lr-scheduler.md](https://github.com/PaddlePaddle/awesome-DeepLearning/files/8085453/11_11.lr-scheduler.md)

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_1 image-augmentation.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+def load_cifar10(is_train, augs, batch_size):
+    dataset = paddle.vision.datasets.Cifar10(mode="train",transform=augs, download=True)
+
+    dataloader = paddle.io.DataLoader(dataset, batch_size=batch_size, num_workers=2, shuffle=is_train)
+
+    return dataloader
+
+```
+
+### 多GPU训练
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_1 image-augmentation.md:167]: 这里需要说明AI Studio只提供一个GPU，所以并不是多GPU训练，可以在本地尝试代码进行多GPU训练。

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_1 image-augmentation.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    if type(m) in [nn.Linear, nn.Conv2D]:
+        nn.initializer.XavierUniform(m.weight)
+
+net.apply(init_weights)
+
+def train_with_data_aug(train_augs, test_augs, net, lr=0.001):
+    train_iter = load_cifar10(True, train_augs, batch_size)
+    test_iter = load_cifar10(False, test_augs, batch_size)
+    loss = nn.CrossEntropyLoss(reduction="none")
+    trainer = paddle.optimizer.Adam(learning_rate=lr,parameters=net.parameters() )
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_1 image-augmentation.md:247]: learning_rate=lr 逗号后空格，net.parameters() 后删掉空格

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_1 image-augmentation.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+我们在CIFAR-10数据集上训练 :numref:`sec_resnet`中的ResNet-18模型。
+回想一下 :numref:`sec_multi_gpu_concise`中对多GPU训练的介绍。
+接下来，我们[**定义一个函数，使用多GPU对模型进行训练和评估**]。
+
+
+
+```python
+#@save
+def train_batch_ch13(net, X, y, loss, trainer, devices):
+    """用多GPU进行小批量训练
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_1 image-augmentation.md:178]: 这个多GPU在本地确认效果ok么？

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_2 fine-tuning.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+train_imgs = paddle.vision.datasets.DatasetFolder(os.path.join(data_dir, 'train'))
+test_imgs = paddle.vision.datasets.DatasetFolder(os.path.join(data_dir, 'test'))
+```
+
+下面显示了前8个正类样本图片和最后8张负类样本图片。
+正如你所看到的，[**图像的大小和纵横比各有不同**]。
+
+
+
+```python
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_2 fine-tuning.md:90]: 这一步在aistudio执行完成后，在显示图像前会print出一些这样的结果：
array([<matplotlib.axes._subplots.AxesSubplot object at 0x7f6206199a90>,
       <matplotlib.axes._subplots.AxesSubplot object at 0x7f620633b4d0>,

这个是正常的么

File: Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_2 fine-tuning.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        params_1x = [param for name, param in net.named_parameters()
+             if name not in ["fc.weight", "fc.bias"]]
+        trainer = paddle.optimizer.SGD(learning_rate=learning_rate, parameters=[{'params': params_1x},
+                                   {'params': net.fc.parameters(),
+                                    'learning_rate': learning_rate * 10}],
+                                    weight_decay=0.001)
+    else:
+        trainer = paddle.optimizer.SGD(learning_rate=learning_rate, parameters=net.parameters(), 
+                                  weight_decay=0.001)
+    d2l.train_ch13(net, train_iter, test_iter, loss, trainer, num_epochs,devices)
```
Comment (ZhangHandi) [inline Dive-into-DL-paddlepaddle/docs/13_computer-vision/13_2 fine-tuning.md:191]: num_epochs, 注意空格

**Comment by w5688414** [comment]
14.7节，结果正确，格式整齐，OK

**Comment by 1649759610** [comment]
def init_weights(m):
    if type(m) == nn.Linear:
        nn.initializer.XavierUniform(m.weight)
    if type(m) == nn.LSTM:
        for param in m.state_dict():
            if "weight" in param:
                print(param)
                nn.initializer.XavierUniform(m.state_dict()[param])
net.apply(init_weights);

这个参数初始化方式，并不能成功修改net网络中的参数，请参考如下赋值方式：
def init_weights(self, layer):
        """ Initialization hook """
        if isinstance(layer, (nn.Linear, nn.Embedding)):
            # In the dygraph mode, use the `set_value` to reset the parameter directly,
            # and reset the `state_dict` to update parameter in static mode.
            if isinstance(layer.weight, paddle.Tensor):
                layer.weight.set_value(
                    paddle.tensor.normal(
                        mean=0.0,
                        std=self.initializer_range
                        if hasattr(self, "initializer_range") else
                        self.bert.config["initializer_range"],
                        shape=layer.weight.shape))
        elif isinstance(layer, nn.LayerNorm):
            layer._epsilon = 1e-12

原链接：https://github.com/PaddlePaddle/PaddleNLP/blob/develop/paddlenlp/transformers/bert/modeling.py

**Comment by ZhangHandi** [comment]
13.2 LGTM

**Comment by ZhangHandi** [comment]
13.1 LGTM

**Comment by 1649759610** [comment]
15.2 review无问题

**Comment by tngt** [comment]
review均通过，可以merge
